walk-in-interview/src/main/java/com/google/firestore/FireStoreUtils.java

```diff
@@ -0,0 +1,72 @@
+package com.google.firestore;
+
+import com.google.api.core.ApiFuture;
+import com.google.auth.oauth2.GoogleCredentials;
+import com.google.cloud.firestore.DocumentReference;
+import com.google.cloud.firestore.DocumentSnapshot;
+import com.google.cloud.firestore.Firestore;
+
+import com.google.cloud.firestore.WriteResult;
+import com.google.firebase.FirebaseApp;
+import com.google.firebase.FirebaseOptions;
+import com.google.firebase.cloud.FirestoreClient;
+
+import java.io.IOException;
+import java.util.concurrent.ExecutionException;
+
+/** Util methods related to Cloud Firestore database. */
+public final class FireStoreUtils {
+
+    private static final String PROJECT_ID = "google.com:walk-in-interview";
+    private static Firestore db;
+
+    private FireStoreUtils() {}
+
+    // Use the application default credentials
+    private static void init() throws IOException {
+        GoogleCredentials credentials = GoogleCredentials.getApplicationDefault();
+        FirebaseOptions options = new FirebaseOptions.Builder()
+                .setCredentials(credentials)
+                .setProjectId(PROJECT_ID)
+                .build();
+        FirebaseApp.initializeApp(options);
+
+        db = FirestoreClient.getFirestore();
+    }
+
+    public static Firestore getFireStore() throws IOException {
+        if (db == null) {
+            init();
+        }
+
+        return db;
+    }
+
+    public static ApiFuture<DocumentReference> store(String collection, Object item)
```

Comment:
write an explanation of what this function does (same comment for other functions below)

Reply:
Fixed